javalib/src/main/scala/java/nio/file/Files.scala

```diff
@@ -624,15 +623,22 @@ object Files {
     try _walkFileTree(start, options, maxDepth, visitor)
     catch { case TerminateTraversalException => start }
 
+  // The sense of how LinkOption follows links or not is somewhat
+  // inverted because of a double negative.  The absense of
+  // LinkOption.NOFOLLOW_LINKS means follow links, the default.
+  // There is no explicit LinkOption.FOLLOW_LINKS.
+  private def linkOptsFromFileVisitOpts(
+      options: Array[FileVisitOption]): Array[LinkOption] = {
+    if (options.contains(FileVisitOption.FOLLOW_LINKS)) Array.empty[LinkOption]
+    else Array(LinkOption.NOFOLLOW_LINKS)
+  }
+
   private def _walkFileTree(start: Path,
                             options: Set[FileVisitOption],
                             maxDepth: Int,
                             visitor: FileVisitor[_ >: Path]): Path = {
-    val stream = walk(start,
-                      maxDepth,
-                      0,
-                      options.toArray.asInstanceOf[Array[FileVisitOption]],
-                      SSet.empty)
+    val optsArray  = options.toArray.asInstanceOf[Array[FileVisitOption]]
```

Comment:
While you're here: is the `asInstanceOf` actually necessary?

Reply:
Unfortunately it is necessary. `toArray` on Java collections results in `Array[Object]` (or `Array[AnyRef` in scala compiler). 
Altough another syntax may be used which is ```options.toArray(new Array[FileVisitOption](options.size))``` 
I can change it, but I think it looks cleaner as it is right now.

Reply:
Ah, it's the Java collections! Then it's worse than I thought. This cast is actually *invalid*, since `toArray` returns an `Array[Object]`, which is *not* an `Array[FileVisitOption]`. It seems that Scala Native lets that go through, but that's clearly undefined behavior and we should not rely on it.

Please switch to the other alternative that you mentioned (with the `new Array`).